using-timescaledb/continuous-aggregates.md

```diff
@@ -16,145 +16,189 @@ every refresh. Only new and/or invalidated data will be calculated.  Since this
-amount of data processed in a single job. If there is still more work to do to
-catch up to where we need to be based on the `refresh_lag` setting, another job
-will be spawned immediately following completion. This means that the part of the
-data processed by previous jobs will be available even before the
-materialization is fully caught up.
+### Advanced Usage
+
+When designing continuous aggregates, we wanted to avoid write
+amplification since this drastically affect insert rate for
+write-heavy time-series workloads. As a result, data is aggregated
+into a materialized hypertable (one for each continuous aggregate
+view) containing partial aggregates trailing the underlying hypertable
+(the table `conditions` in the example above).
+
+Each continuous aggregate has a *bucket width* that comes from the
+`time_bucket` call in the view definition. This is the granularity
+that is updated and typically new data is added to the last bucket. By
+not updating the last bucket of the view, we can avoid having to
+refresh the last bucket multiple times as new data comes in. Buckets
+older than that will be aggregated, but since new data does not
+(normally) arrive into those bucket, an aggegate for the bucket is
+computed only once, which avoid write amplification.
+
+However, when querying the continuous aggregate view (for example,
+`conditions_summary_daily`) you typically want to get a complete view
+of the data, including the data in the underlying table, so the view
+will get one part of the result from the materialized hypertable and
+one part from the underlying table.
+
+If you, however, want to just get the data from the materialized
+hypertable, and not include data from the underlying table, you can
+set the option `timescaledb.materialized_only` to `true` using `ALTER
+VIEW`:
+
+```sql
+ALTER VIEW conditions_summary_hourly SET (
+    timescaledb.materialized_only = true
+);
+```
+
+As noted above, the materialized hypertable lag behind the underlying
+table by one bucket width (parameter `bucket_width` in the
+`time_bucket` function of the view). However, you can change this lag
+to either increase it (or decrease it) by setting the value for the
+`timescaledb.refresh_lag` option of the view and the background worker
+will attempt to materialize everything before `refresh_lag` +
+`bucket_width`.
+
+For example, using `conditions_summary_hourly` above, the bucket width
+is 1 hour, so if we set the `refresh_lag` to 1 hour, the
+materialization will be 2 hours behind the current time.
+
+```sql
+ALTER VIEW conditions_summary_hourly SET (
+  timescaledb.refresh_lag = '1 hour'
+);
+```
+
+Lowering the `refresh_lag` parameter means that the aggregates follow
+inserts more closely, but can cause write amplification that might
+degrade insert performance. You can also set `refresh_lag` to negative
+values, so if you set `refresh_lag` to `-bucket_width` will make the
+continuous aggregate read all data as it is added.
```

Comment:
Again, this doesn't seem up-to-date with real-time aggs. 

If I set a large `refresh_lag`, does that mean that a larger portion of the query on the view will be fetched using the "raw hypertable"?  I guess this is lost in the discussion about write amplification.

Reply:
I see no changes here.

To clarify, if I set `refresh_lag=-bucket_width` and `timescaledb.materialized_only = false`, does that mean that the "raw" part of the union query  effectively returns no tuples because everything is materialized?

This relationship isn't clear.

It seems to me that `refresh_lag` determines the size of the interval covered by the "raw" query when this is a real-time aggregate. This seems like an important detail that is lost.

Reply:
Added a paragraph about it.